### ISSUE TITLE
ClassCastException: Report-processing with "HTML-table-structures" cause exception if the structure contains comments

### ISSUE THREAD

**Issue by speckyspooky** [eclipse-birt/birt#1812]
If a report use dynamic data-content with html-table-structures
then a class cast exception can occur if the structure contains html-comments.

In my reports I used entered user text (=html-text) which can contains according html-structures:

The following snippet will cause the exception:
```
<table border=1>
<!-- comment: line break -->
<tbody>
<tr><td>standard text</td></tr>
</tbody>
</table>
```

![html-table-processing](https://github.com/user-attachments/assets/b95f757b-7bf2-47f7-ac10-c8e5279aeb33)

The reason of it is that the table-processor doesn't check the XML-node-type correctly.

I will provide a fix for that.

**Comment by speckyspooky** [comment]
The fix is added on PR #1813

**Comment by speckyspooky** [comment]
The fix is merged to the master branch.
